library/src/otx/data/factory.py

```diff
@@ -41,58 +44,71 @@ class OTXDatasetFactory:
 
     @classmethod
     def create(
-        cls: type[OTXDatasetFactory],
+        cls,
         task: OTXTaskType,
-        dm_subset: DmDataset,
+        dm_subset: DmDataset | DatasetNew,
```

Comment:
Is it for backward compatibility? I don't think we need that.  cc @leoll2 
Same for "convert_from_legacy".

GetiTune will depend on the new Datumaro Dataset, Geti classic has no dependency on the develop branch. OTX uses file system, not datumaro directly. I propose to support only the latest version if no stoppers.

Reply:
Legacy datumaro datasets can still be passed here since the conversion to the new datumaro dataset schema happens in this factory. We will remove this in the future, but as long as we still support the legacy datasets in datumaro it also makes sense to keep it here. If you already have a new datumaro Dataset, you can initialize the tasks dataset immediately. 

It also reduces the scope for this PR a bit.